catalystneuro/neuroconv
# Swap to Pydantic for schema validation

fix #261

## Motivation

Showcase of how to use more rigorous Pydantic model validation instead of `jsonschema` validation.

Also moving it to the interface level, which will fix the current issue of interfaces not validating any schemas when used for stand-alone write to an NWB file.

This PR is only for `source`, subsequent PRs will extend to the other schemas.



## How to test the behavior?

Theoretically already tested but can add a few explicit tests to verify our understanding of the behavior going forward.


## Checklist

- [ ] Have you thoroughly read our [Developer Guide](https://neuroconv.readthedocs.io/en/main/developer_guide.html)?
- [ ] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/catalystneuro/neuroconv/pulls) for the same change?
- [ ] If this PR fixes an issue, is the first line of the PR description `fix #XXX` where `XXX` is the issue number?
- [ ] Did you update the [CHANGLOG.md](https://github.com/catalystneuro/neuroconv/blob/main/CHANGELOG.md) file on your branch to describe your changes?
